### DISCUSSION TITLE
Source code view in docs

### DISCUSSION THREAD

**Discussion by renefritze** [pymor/pymor#1241] [Ideas]
Currently we have `sphinx.ext.viewcode` enabled when building our sphinx docs. This puts a link next to each documented entity to show a html view of the entity's source (highlighted by pygments). I'm looking for some general feedback from our documentation users if this is useful, distracting, whatever.

**Comment by renefritze** [comment]
Additionally, I noticed an oddity in the source view:
https://docs.pymor.org/2020.2.0/_modules/pymor/tools/deprecated.html#Deprecated
Something inserts those `[docs` makers which screws up indentation for me.

**Comment by sdrave** [comment]
I find that link quite useful. Not so much for pyMOR, since I always have the source at hand, but for other projects.

**Reply by renefritze** [reply]
How would think about this if that link sends you to the project's repo view (github) instead?

**Reply by TiKeil** [reply]
I agree, I also like to click on the source code link. I don't mind where it puts me though, so the repo view would also be ok for me.

**Reply by pmli** [reply]
> How would think about this if that link sends you to the project's repo view (github) instead?

I like this idea, I used it quite a bit with other projects.

**Reply by sdrave** [reply]
I'd be fine with github links. Is there a plugin which does that?

**Reply by renefritze** [reply]
Kinda sorta. There's an unmaintained one might work. The Sphinx-rtd theme inserts links as well. `sphinx.ext.viewcode` conflicts with sphinx-autoapi atm, so just replacing the link targets there and repacking isn't really an option if I don't get the conflict sorted out.
